data/transactions/logic/assembler.go

```diff
@@ -1764,29 +1806,69 @@ func prepareVersionedPseudoTable(version uint64) map[string]map[int]OpSpec {
 	return m
 }
 
-type lineError struct {
-	Line int
-	Err  error
+type sourceError struct {
+	Line   int
+	Column int
+	Err    error
 }
 
-func (le lineError) Error() string {
-	return fmt.Sprintf("%d: %s", le.Line, le.Err.Error())
+func (se sourceError) Error() string {
+	if se.Column != 0 {
+		return fmt.Sprintf("%d:%d: %s", se.Line, se.Column, se.Err.Error())
+	}
+	return fmt.Sprintf("%d: %s", se.Line, se.Err.Error())
 }
 
-func (le lineError) Unwrap() error {
-	return le.Err
+func (se sourceError) Unwrap() error {
+	return se.Err
+}
+
+func errorLinef(line int, format string, a ...interface{}) *sourceError {
+	return &sourceError{line, 0, fmt.Errorf(format, a...)}
+}
+
+func typecheck(expected, got StackType) bool {
+	// Some ops push 'any' and we wait for run time to see what it is.
+	// Some of those 'any' are based on fields that we _could_ know now but haven't written a more detailed system of typecheck for (yet).
+	if expected == StackAny && got == StackNone { // Any is lenient, but stack can't be empty
+		return false
+	}
+	if (expected == StackAny) || (got == StackAny) {
+		return true
+	}
+	return expected == got
+}
+
+type token struct {
+	str  string
+	col  int
+	line int
```

Comment:
some ambiguous usages of "line" as integer representing the line count and "line" as a string at that line

Reply:
It seems like this is the only place where `line` means text:
```
func tokensFromLine(line string, lineno int) []token {
```
I could change it, but I don't love calling it `lineStr` or something like that.  What do you think.

(There was also a 3 line piece of disassemble that used `line` as a string, I made that variable `instruction.)

Reply:
I think of `__LINE__` and that is a number, how about for the other one `source`, `sourceline`, `linetext`